### DISCUSSION TITLE
Microsoft Azure Deployment Stack for Langfuse v3

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#4647] [Ideas]
### Describe the feature or potential improvement

Currently it is recommended to use [Helm/Kubernetes](https://langfuse.com/self-hosting/kubernetes-helm) to deploy Langfuse v3. 

Let's collect interest for an Azure specific deployment stack in this thread. Please share what you'd like to see and whether you are interested in contributing to this.

## EDIT 24.04.2025

We just published https://github.com/langfuse/langfuse-terraform-azure which should deploy Langfuse v3 on Azure using as many native services as possible. We'd love to get your feedback on this 🚀

**Comment by stphnthiel** [comment]
It seems our self-deployed version has stopped working since an hour ago...

```
2024-12-09T12:59:30.34668  Connecting to the container 'langfuse-g47pr27xgb7ns-app'...
2024-12-09T12:59:30.36472  Successfully Connected to container: 'langfuse-g47pr27xgb7ns-app' [Revision: 'langfuse-g47pr27xgb7ns-app--tfngk94-556f45ffbb-hvj28', Replica: 'langfuse-g47pr27xgb7ns-app--tfngk94']
2024-12-09T12:56:40.317867155Z Error: CLICKHOUSE_URL is not configured. Migrating from V2? Check out migration guide at https://langfuse.com/docs/deployment/v3/migrate-v2-to-v3.
```

**Reply by simicvm** [reply]
yeap, because v3 is completely different and you probably had autoupdate turned on...

**Reply by marcklingen** [reply]
+1, it is recommended to pin the major version when running Langfuse self-hosted, see v2 deployment docs: https://langfuse.com/self-hosting/v2/deployment-guide#deploying-the-application

```
docker pull langfuse/langfuse:2
```

You should be able to rollback to `:2` easily, let me know if you have any questions/issues

**Reply by stphnthiel** [reply]
Unfortunately, I *simply* used the Azure template which is pinned to `latest`. Currently investigating how to downgrade.

**Reply by stphnthiel** [reply]
Luckily, that was fairly easy (created a new revision of the container app using `:2`)

Still, quite an effective way to convince users to move to the cloud

**Reply by Steffen911** [reply]
I created a PR at the Azure sample to pin `:2`: https://github.com/Azure-Samples/langfuse-on-azure/pull/18. We're sorry for your experience here, but unfortunately it's hard to never break a `:latest` image. Using a major-pinned version should remove the upgrade risks for self-hosted deployments.

**Reply by marcklingen** [reply]
> Still, quite an effective way to convince users to move to the cloud

That is not the goal here, I am sorry about this! We do not maintain this azure sample and it did not follow our self-host deployment guide of pinning v2. Thanks @Steffen911 for creating the PR on the Azure sample. We plan to create a new Azure sample for v3 (tracking this here), will try to align with the maintainer at Microsoft on this!

**Reply by stphnthiel** [reply]
Also, I was merely joking here. I understood the template was provided *as is* and that responsibility is sort of the deal in self-hosting. 

For us at this stage, we rather have a managed deployment and happily pay for it (incl. the added features). Also you provide a good tier to get started. 

Thanks for the great work and for making parts of LF OSS! And congrats on the new release also!

**Reply by marcklingen** [reply]
thanks @stphnthiel for the feedback!

**Comment by marcklingen** [comment]
Open issue for this in azure-samples: https://github.com/Azure-Samples/langfuse-on-azure/issues/14

@pamelafox, @JungeAlexander, We would love to work with you on this if you are interested as v3 is stable since today's 3.0.0 release including all necessary docs.

**Reply by JungeAlexander** [reply]
Awesome, happy to give feedback and test along the way. And congrats to v3!

**Comment by Steffen911** [comment]
Hey all,
We just published https://github.com/langfuse/langfuse-terraform-azure which should deploy Langfuse v3 on Azure using as many native services as possible. We'd love to get your feedback on this 🙌
